MathLib/LinAlg/Eigen/EigenLinearSolver.cpp

```diff
@@ -32,16 +32,19 @@ class EigenDirectLinearSolver final : public T_BASE
     explicit EigenDirectLinearSolver(EigenMatrix::RawMatrixType &A) : _A(A)
     {
         INFO("-> initialize with the coefficient matrix");
-        _solver.compute(A);
-        if(_solver.info()!=Eigen::Success) {
-            ERR("Failed during Eigen linear solver initialization");
-            return;
-        }
     }
 
     void solve(EigenVector::RawVectorType &b, EigenVector::RawVectorType &x, EigenOption &/*opt*/) override
     {
         INFO("-> solve");
+        if (!_A.isCompressed())
+            _A.makeCompressed();
+        _solver.compute(_A);
```

Comment:
Maybe someone does not always want to always compute a new decomposition of A for each solve step?